rust/agama-lib/src/store.rs

```diff
@@ -17,19 +18,19 @@ use zbus::Connection;
 /// This struct uses the default connection built by [connection function](super::connection).
 pub struct Store<'a> {
     users: UsersStore<'a>,
-    network: NetworkStore<'a>,
+    network: NetworkStore,
     product: ProductStore<'a>,
     software: SoftwareStore<'a>,
     storage: StorageStore<'a>,
     localization: LocalizationStore<'a>,
 }
 
 impl<'a> Store<'a> {
-    pub async fn new(connection: Connection) -> Result<Store<'a>, ServiceError> {
+    pub async fn new(connection: Connection, client: Client) -> Result<Store<'a>, ServiceError> {
```

Comment:
What is `client: Client`? A little bit generic. I would rename the parameter to `http_client`.